pkg/agent/route/route_windows.go

```diff
@@ -330,9 +360,105 @@ func (c *Client) UnMigrateRoutesFromGw(route *net.IPNet, linkName string) error
 
 // Run is not supported on Windows and returns immediately.
 func (c *Client) Run(stopCh <-chan struct{}) {
+	klog.Infof("Starting NetNat and route sync, with sync interval %v", syncInterval)
+	wait.Until(c.syncIPInfra, syncInterval, stopCh)
+}
+
+// syncIPInfra is idempotent and can be safely called on every sync operation.
+func (c *Client) syncIPInfra() {
+	if err := c.syncRoute(); err != nil {
+		klog.Errorf("Failed to sync route: %v", err)
+	}
+
+	if c.proxyAll {
+		if err := c.syncNetNat(); err != nil {
+			klog.Errorf("Failed to sync netNat: %v", err)
+		}
+	}
+
+	klog.V(3).Infof("Successfully synced Node netNat and route")
 }
 
-func (c *Client) listRoutes() (map[string]*util.Route, error) {
+func (c *Client) syncRoute() error {
+	routeMap, err := c.listIPRoutesOnGW()
+	if err != nil {
+		return err
+	}
+
+	restoreRoute := func(route *util.Route) bool {
+		r, ok := routeMap[route.DestinationSubnet.String()]
+		if ok && route.Equal(*r) {
+			return true
+		}
+		if err := util.ReplaceNetRoute(route); err != nil {
+			klog.Errorf("Failed to add route to the gateway: %v", err)
+			return false
+		}
+		return true
+	}
+	c.nodeRoutes.Range(func(_, v interface{}) bool {
+		route := v.(*util.Route)
+		return restoreRoute(route)
+	})
+	if c.proxyAll {
+		c.serviceRoutes.Range(func(_, v interface{}) bool {
+			route := v.(*util.Route)
+			return restoreRoute(route)
+		})
+	}
+	// These routes are installed automatically by the kernel when the address is configured on
+	// the interface. If these routes are deleted manually by mistake, we restore them.
+	gwAutoconfRoutes := []*util.Route{
+		{
+			LinkIndex:         c.nodeConfig.GatewayConfig.LinkIndex,
+			DestinationSubnet: util.NewIPNet(c.nodeConfig.GatewayConfig.IPv4),
+			GatewayAddress:    c.nodeConfig.GatewayConfig.IPv4,
+			RouteMetric:       util.MetricDefault,
+		},
+		{
+			LinkIndex:         c.nodeConfig.GatewayConfig.LinkIndex,
+			DestinationSubnet: c.nodeConfig.PodIPv4CIDR,
+			GatewayAddress:    c.nodeConfig.GatewayConfig.IPv4,
+			RouteMetric:       util.MetricDefault,
+		},
+	}
+
+	for _, route := range gwAutoconfRoutes {
+		restoreRoute(route)
+	}
+	return nil
+}
+
+func (c *Client) syncNetNat() error {
+	if err := util.NewNetNat(antreaNatNodePort, virtualNodePortDNATIPv4Net); err != nil {
+		return err
+	}
+
+	restoreNetNatStaticMapping := func(m *util.NetNatStaticMapping) bool {
+		if err := util.ReplaceNetNatStaticMapping(m); err != nil {
+			klog.Errorf("Failed to add netNat static mapping: %v", err)
+			return false
+		}
+		return true
+	}
+
+	c.netNat.Range(func(_, v interface{}) bool {
+		mapping := v.(*util.NetNatStaticMapping)
+		return restoreNetNatStaticMapping(mapping)
+	})
+
+	return nil
+}
+
+func (c *Client) isServiceRoute(route *util.Route) bool {
+	// If the gateway IP is the virtual Service IP , then it is a route which is added by AntreaProxy.
+	if route.GatewayAddress.Equal(config.VirtualServiceIPv4) {
```

Comment:
Is it possible GatewayAddress is nil?

Reply:
It is impossible since it is not a pointer type.

Reply:
Currently it is impossible, but I think we can add a check in case that it might be nil in the future.